app/controllers/passengers_controller.rb

```diff
@@ -48,8 +48,12 @@ def destroy
   private
 
   def passenger_params
-    params.require(:passenger).permit(:name, :address, :email, :phone,
-      :wheelchair, :active, :permanent, :expiration, :note)
+    permitted_params = params.require(:passenger).permit(:name, :address, :email, :phone,
+        :wheelchair, :active, :permanent, :expiration, :note)
+    unless @current_user.admin?
+      permitted_params = permitted_params.except(:active, :permanent)
+      return permitted_params
```

Comment:
This should go at the end of the method, not the end of the conditional. I also would not write `return`. Just `permitted_params`.